doctest/parts/doctest_fwd.h

```diff
@@ -476,6 +476,36 @@ namespace doctest {
 
 DOCTEST_INTERFACE extern bool is_running_in_test;
 
+#if DOCTEST_MSVC
```

Comment:
this is a Windows-specific problem, so builds with mingw or clang should also be affected. Perhaps the correct check here is `DOCTEST_PLATFORM_WINDOWS`